src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -558,8 +558,11 @@ private int LookupAggregatorStoreForDeltaWithReclaim(KeyValuePair<string, object
         {
             if (length > 1)
             {
-                // Note: We are using storage from ThreadStatic, so need to make a deep copy for Dictionary storage.
-                // Create or obtain new arrays to temporarily hold the sorted tag Keys and Values
+                // Create or obtain new arrays to temporarily hold the sorted tag Keys and Values.
```

Comment:
this fixed an incorrect comment (there was no deep copy here.), and also expanded further on "why"... Still unsure, if folks find such code comments useful or distracting. Let me know.

Reply:
Did you mean to include these on this PR? Feel like they should be their own thing.